ILEditing/MechanicILChanges.cs

```diff
@@ -835,16 +858,29 @@ private static void ChangeWaterQuadColors(ILContext il)
             cursor.Emit(OpCodes.Ldloc, 8);
             cursor.Emit(OpCodes.Ldloc, 3);
             cursor.Emit(OpCodes.Ldloc, 4);
+            
+            // Caching these values can save a LOT of overhead at runtime.
+            ModWaterStyle sunkenSeaWater = ModContent.Find<ModWaterStyle>("CalamityMod/SunkenSeaWater");
+            ModWaterStyle sulphuricWater = ModContent.Find<ModWaterStyle>("CalamityMod/SulphuricWater");
+            ModWaterStyle sulphuricDepthsWater = ModContent.Find<ModWaterStyle>("CalamityMod/SulphuricDepthsWater");
+            ModWaterStyle upperAbyssWater = ModContent.Find<ModWaterStyle>("CalamityMod/UpperAbyssWater");
+            ModWaterStyle middleAbyssWater = ModContent.Find<ModWaterStyle>("CalamityMod/MiddleAbyssWater");
+            ModWaterStyle voidWater = ModContent.Find<ModWaterStyle>("CalamityMod/VoidWater");
```

Comment:
Are these not cached in the current implementation? Shouldn't they be stored outside the function, because it'll never change during a mod load cycle?

Reply:
I'm unfamiliar with the codebase Calamity has, so I'll need to see if there's a better way to access them. I did it this way as it seemed the least intrusive and the easiest to do.

Reply:
Storing them is pointless, they could be constructor-initialized before the content has been registered.